### DISCUSSION TITLE
Is vendorIdentifier a DeviceId?

### DISCUSSION THREAD

**Discussion by mayeranalytics** [JoelBender/bacpypes#429] [Q&A]
The `LocalDeviceObject` class has (resp. uses) a `vendorIdentifier` parameter. Is this `vendorIdentifier` the `deviceId` in the Bacnet sense?

**Comment by JoelBender** [comment]
No, it's an unsigned integer that is assigned to the vendor (by ASHRAE) that allows them to provide custom/proprietary objects and properties and still use BACnet concepts and encodings to exchange data.  There are lots of different design patterns, like...

- extending standard objects with custom properties
- extending some enumerations with custom values
- providing custom objects with standard or custom properties

So if you know the vendor identifier you have a way to look up their "schema", what objects and properties they provide and perhaps some additional information like if the property is writable or commandable, and what values are appropriate.

The list of assigned vendor identifiers is on the [BACnet SSPC web site](http://www.bacnet.org/VendorID/BACnet%20Vendor%20IDs.htm), along with a summary of [their countries](http://www.bacnet.org/VendorID/index.html) (and many of them are inter-/multi-national).

If you are creating a product that supports BACnet I would encourage you to write to the [ASHRAE Manager of Standards](http://www.bacnet.org/DL-Docs/Procedures-Vendor-ID-rev3-15-2012.pdf) and obtain a vendor identifier.  There is no charge, and even if you don't have any custom content yet, having the vendor identifier is such a benefit to customers and asset management (along with software and firmware version numbers please!) that we made it required :-).
